Rubberduck.UnitTesting/UnitTesting/TestEngine.cs

```diff
@@ -34,16 +34,26 @@ internal class TestEngine : ITestEngine
         private readonly IUiDispatcher _uiDispatcher;
         private readonly IVBE _vbe;
 
-        private readonly List<TestMethod> LastRun = new List<TestMethod>();
-        private readonly Dictionary<TestOutcome, List<TestMethod>> resultsByOutcome = new Dictionary<TestOutcome, List<TestMethod>>();
-        public IEnumerable<TestMethod> Tests { get; private set; }
+        private Dictionary<TestMethod, TestOutcome> _knownOutcomes = new Dictionary<TestMethod, TestOutcome>();
+        private List<TestMethod> _lastRun = new List<TestMethod>();
+        private List<TestMethod> _tests;
+
+        public IEnumerable<TestMethod> Tests => _tests;
+
+        public IReadOnlyList<TestMethod> LastRunTests => _lastRun;
+
         public bool CanRun => AllowedRunStates.Contains(_state.Status) && _vbe.IsInDesignMode;
-        public bool CanRepeatLastRun => LastRun.Any();
+        public bool CanRepeatLastRun => _lastRun.Any();
         
-        private bool refreshBackoff;
-
+        private bool _refreshBackoff;
```

Comment:
TBH the name of the variable doesn't make sense - "turn refresh back off" is what I think of. Is that what it means?

Reply:
I must have looked at that a hundred times and and said "huh"? It means "don't refresh from the parser if this flag is set and it's otherwise ready".  Any suggestions? Maybe `_deferRefresh` or `_refreshUnsafe`?

Reply:
I probably would flip the logic and call it `_reactToParserStates` or more simply (but less descriptively), `_listening`.